### PR TITLE
Add Functionality For Multiple Devfile Name Variations

### PR DESCRIPTION
**Pull request by Jdubrick**
### What does this PR do?:

This PR updates the `DownloadGitRepoResources` function to allow for different naming schemes for devfiles to be supported. Before the change only `devfile.yaml` was being recognized. This PR adds support for `devfile.yaml`, `devfile.yml`, `.devfile.yaml` and `.devfile.yml`.  In addition to this it allows for more naming schemes to be added in the future.

### Which issue(s) this PR fixes:

fixes https://github.com/devfile/api/issues/1362

### PR acceptance criteria:
Testing and documentation do not need to be complete in order for this PR to be approved. We just need to ensure tracking issues are opened and linked to this PR, if they are not in the PR scope due to various constraints.

- [ ] Unit/Functional tests

  

- [ ] [QE Integration test](https://github.com/devfile/integration-tests) 

  

- [ ] Documentation (READMEs, Product Docs, Blogs, Education Modules, etc.)

   

- [ ] Client Impact

  

- [ ] Gosec scans
  

### How to test changes / Special notes to the reviewer:
Running `make test` should pass with the addition of the new tests for the devfile validation.

### CODE REVIEW

File: pkg/devfile/parser/util/utils.go [hunk 1]
```diff
@@ -25,10 +25,8 @@ import (
 	"github.com/hashicorp/go-multierror"
 )
 
-// Default filenames for create devfile
-const (
-	OutputDevfileYamlPath = "devfile.yaml"
-)
+// Contains common naming conventions for devfiles to look for when downloading resources
+var DevfilePossibilities = [...]string{"devfile.yaml", "devfile.yml", ".devfile.yaml", ".devfile.yml"}
```
Comment (thepetk) [inline pkg/devfile/parser/util/utils.go:29]: I think this sorting is different than the one mentioned in https://github.com/devfile/devfile-web/blob/main/libs/docs/src/docs/no-version/library.md#procedure, when we first added the parser support of the `.y(a)ml` (PR https://github.com/devfile/library/pull/186/files). I think the correct sort should be:

```suggestion
var DevfilePossibilities = [...]string{"devfile.yaml", ".devfile.yaml", "devfile.yml", ".devfile.yml"}
```

**Review by thepetk — CHANGES_REQUESTED** [review]
@Jdubrick a minor update proposed from my side regarding the sorting that the  `ValidateDevfileExistence` is following.

**Comment by thepetk** [comment]
> ## [Codecov](https://app.codecov.io/gh/devfile/library/pull/198?src=pr&el=h1&utm_medium=referral&utm_source=github&utm_content=comment&utm_campaign=pr+comments&utm_term=devfile) Report
> Attention: `2 lines` in your changes are missing coverage. Please review.
> 
> > Comparison is base [(`2a1d045`)](https://app.codecov.io/gh/devfile/library/commit/2a1d045e1e65ffe04ebbe598c56582eea794dc5a?el=desc&utm_medium=referral&utm_source=github&utm_content=comment&utm_campaign=pr+comments&utm_term=devfile) 60.83% compared to head [(`22c637b`)](https://app.codecov.io/gh/devfile/library/pull/198?src=pr&el=desc&utm_medium=referral&utm_source=github&utm_content=comment&utm_campaign=pr+comments&utm_term=devfile) 60.88%.
> 
> [Files](https://app.codecov.io/gh/devfile/library/pull/198?src=pr&el=tree&utm_medium=referral&utm_source=github&utm_content=comment&utm_campaign=pr+comments&utm_term=devfile)	Patch %	Lines
> [pkg/devfile/parser/util/mock.go](https://app.codecov.io/gh/devfile/library/pull/198?src=pr&el=tree&utm_medium=referral&utm_source=github&utm_content=comment&utm_campaign=pr+comments&utm_term=devfile#diff-cGtnL2RldmZpbGUvcGFyc2VyL3V0aWwvbW9jay5nbw==)	0.00%	[1 Missing ⚠️ ](https://app.codecov.io/gh/devfile/library/pull/198?src=pr&el=tree&utm_medium=referral&utm_source=github&utm_content=comment&utm_campaign=pr+comments&utm_term=devfile)
> [pkg/devfile/parser/util/utils.go](https://app.codecov.io/gh/devfile/library/pull/198?src=pr&el=tree&utm_medium=referral&utm_source=github&utm_content=comment&utm_campaign=pr+comments&utm_term=devfile#diff-cGtnL2RldmZpbGUvcGFyc2VyL3V0aWwvdXRpbHMuZ28=)	85.71%	[1 Missing ⚠️ ](https://app.codecov.io/gh/devfile/library/pull/198?src=pr&el=tree&utm_medium=referral&utm_source=github&utm_content=comment&utm_campaign=pr+comments&utm_term=devfile)
> Additional details and impacted files
> [☔ View full report in Codecov by Sentry](https://app.codecov.io/gh/devfile/library/pull/198?src=pr&el=continue&utm_medium=referral&utm_source=github&utm_content=comment&utm_campaign=pr+comments&utm_term=devfile). 📢 Have feedback on the report? [Share it here](https://about.codecov.io/codecov-pr-comment-feedback/?utm_medium=referral&utm_source=github&utm_content=comment&utm_campaign=pr+comments&utm_term=devfile).

Not related to this PR, but I think we can add more failure test cases to the `Test_DownloadGitRepoResources` so we can include more cases inside the `DownloadGitRepoResources` func.

File: pkg/devfile/parser/util/utils.go (hunk 1 above)
Comment (Jdubrick) [reply pkg/devfile/parser/util/utils.go]: Thanks for pointing that out. For the functionality it doesn't change anything but could improve search performance slightly if that is the order we want to search in (and if they are most commonly named in that order). See my last commit for the change :)

**Review by thepetk — APPROVED** [review]
/lgtm

**Pull request merged**